relay-cabi/src/processing.rs

```diff
@@ -220,3 +221,22 @@ pub unsafe extern "C" fn relay_parse_release(value: *const RelayStr) -> RelayStr
     let release = sentry_release_parser::Release::parse((*value).as_str())?;
     RelayStr::from_string(serde_json::to_string(&release)?)
 }
+
+/// Validate a dynamic rule condition.
+/// We only validate conditions in Relay since conditions are the only part of the rules that are
+/// volatile and need constant updating to keep in sync between Relay and Sentry
+#[no_mangle]
+#[relay_ffi::catch_unwind]
+pub unsafe extern "C" fn relay_validate_dynamic_rule_condition(value: *const RelayStr) -> RelayStr {
+    let ret_val = match serde_json::from_str::<RuleCondition>((*value).as_str()) {
+        Ok(condition) => {
+            if condition.supported() {
+                "".to_string()
```

Comment:
You can return `RelayStr::new("")` here. No allocation required.

Reply:
I know, but I don't like it.
If you **insist** I'll do it but here are my reasons not to:

What I want to do (in a future PR) is change the current RealyStr::new("") to:
```
   pub(crate) fn new(s: &'static str) -> RelayStr 
```
and
```
   pub(crate) fn unsafe unsafe_new(s: &str) -> RelayStr 
```

That also affects the implementation of ` impl<'a> From<&'a str> for RelayStr` so I need to look
a bit more in detail at it.

Until then I consider the implementation of `RealyStr::new` as broken, it is unsafe and breaks the Rust
guarantees. It will happily compile something like:

```
pub fn bad_function() -> RelayStr {
    RelayStr::new(&format!("Hello {}", "Jan"))
}
```

Reply:
@jan-auer i feel like we should revisit this string pattern. I remember we incorrectly deduced what the allocation / borrow behavior once before.

Reply:
@mitsuhiko Right, we had bugs prior to Relay in libsymbolic. It's worth noting that the Python layer **always** copies the string anyway. 

@RaduW you're right, the name `new` doesn't carry the implications the function has. It is worth noting that `RelayStr` is only to be used in an unsafe FFI layer, so the exposure is pretty low. Let's use `RelayStr::default` for now, which will inherently not have any borrowing problems.

To continue a discussion from Slack: There is not much code, and it is easy to enumerate _all_ places where we use `RelayStr::new`:
- `relay_data_category_name`, which always returns static strings
- `RelayStr::default`, which can be considered an implementation detail of empty strings in `RelayStr`
- `relay_str_from_cstr`, which deals with pointers. It's also dead code
- `relay_validate_pii_config`, which can also be changed to `RelayStr::default`
- `relay_convert_datascrubbing_config`, which returns a static `"{}"`.

To sum this up, we can address this much simpler by:
- Changing the signature to `new(s: &'static str) -> Self` (this is now inherently safe)
- Removing `relay_str_from_cstr`